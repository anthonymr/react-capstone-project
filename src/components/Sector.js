import { NavLink } from 'react-router-dom';
import { PropTypes } from 'prop-types';

function Sector({ sector, change }) {
  const url = encodeURI(`sectors/${sector}`);
  return (
    <NavLink to={url}>
      <h1>{sector}</h1>
      <span>{change}</span>
    </NavLink>
  );
}

Sector.propTypes = {
  sector: PropTypes.string.isRequired,
  change: PropTypes.string.isRequired,
};

export default Sector;
