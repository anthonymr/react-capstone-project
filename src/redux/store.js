import { configureStore } from '@reduxjs/toolkit';
import sectorsReducer from './sectors/sectorsSlice';

const store = configureStore({
  reducer: {
    sectors: sectorsReducer,
  },
});

export default store;
