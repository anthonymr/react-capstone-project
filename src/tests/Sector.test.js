import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Sector from '../components/Sector';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Provider store={store}>
          <Sector sector="Basic Materials" change="0.75157%" />
        </Provider>,
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
