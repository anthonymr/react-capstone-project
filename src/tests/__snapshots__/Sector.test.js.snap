// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <a
    className=""
    href="/sectors/Basic%20Materials"
    onClick={[Function]}
  >
    <h1>
      Basic Materials
    </h1>
    <span>
      0.75157%
    </span>
  </a>,
  ",",
]
`;
